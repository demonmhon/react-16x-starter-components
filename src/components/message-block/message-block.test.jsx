import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import MessageBlock from './index';

describe('MessageBlock component', () => {
  test('it matches the snapshot', () => {
    const component = create(<MessageBlock />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('use the given childen as MessageBlock text', () => {
    const MessageBlockText = 'OK';
    const component = shallow(<MessageBlock>{MessageBlockText}</MessageBlock>);
    expect(component.text()).toEqual(MessageBlockText);
  });

  test('has the provided class name', () => {
    const className = 'the-custom-css-class';
    const component = shallow(<MessageBlock className={className} />);
    expect(component.hasClass(className)).toBe(true);
  });

  describe('type', () => {
    test('set the type with valid type value', () => {
      const typeName = MessageBlock.Type.Error;
      const expectedClassName = 'message-block--type-error';
      const component = shallow(<MessageBlock type={typeName} />);
      expect(component.hasClass(expectedClassName)).toBe(true);
    });

    test('not set an invalid type', () => {
      const typeName = 'invalidType';
      const expectedClassName = 'message-block';
      const component = shallow(<MessageBlock type={typeName} />);
      expect(component.hasClass(expectedClassName)).toBe(true);
    });
  });
});
