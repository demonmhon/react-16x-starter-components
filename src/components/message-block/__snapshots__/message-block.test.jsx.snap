// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBlock component it matches the snapshot 1`] = `
<div
  className="message-block message-block--type-info"
/>
`;
